src/mlpack/methods/ann/augmented/tasks/score_impl.hpp

```diff
@@ -0,0 +1,99 @@
+/**
+ * @file score_impl.hpp
+ * @author Konstantin Sidorov
+ *
+ * Implementation of scoring functions
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_AUGMENTED_TASKS_SCORE_IMPL_HPP
+#define MLPACK_METHODS_AUGMENTED_TASKS_SCORE_IMPL_HPP
+
+// In case it hasn't been included yet.
+#include "score.hpp"
+
+namespace mlpack {
+namespace ann /* Artificial Neural Network */ {
+namespace augmented /* Augmented neural network */ {
+namespace scorers /* Scoring utilities for augmented */ {
+
+double SequencePrecision(arma::field<arma::colvec> trueOutputs,
+                         arma::field<arma::colvec> predOutputs)
+{
+  double score = 0;
+  auto testSize = trueOutputs.n_elem;
+  assert(testSize == predOutputs.n_elem);
+
+  for (size_t i = 0; i < testSize; i++)
+  {
+    auto prediction = trueOutputs.at(i);
+    auto output = predOutputs.at(i);
+
+    bool ok = true;
+
+    if (output.n_elem != prediction.n_elem)
+    {
+      ok = false;
+    }
+    else
+    {
+      for (size_t j = 0; j < prediction.n_elem; ++j) {
```

Comment:
The only problem I see is, that ``approx_equal``was introduced in Version 6.700, but we support >= 4.200.0, so I guess what we could do here is to backport the function as we did for the e.g. ``ind2sub`` in ``core/arma_extend``.